### DISCUSSION TITLE
ALOS_merge and p2p_processing.csh Not Working

### DISCUSSION THREAD

**Discussion by x3zou** [gmtsar/gmtsar#717] [Q&A] [answered]
My dear expert fellows,
  I'm trying to merge 4 ALOS2 strip-mode frames with [ALOS_merge] and use [p2p_processing.csh] to process the merged data. 
  The way I merge it is that, after pre-processing all the data, I merged file1 + file2 = merge12 then merge12 + file3 = merge123, then merge123 + file4 = merge1234. 
  For convenience, I renamed the ALOS_merge results as below and keep them into the raw folder:
 IMG-HH-ALOS242498xxxx-220406-FBDR1.1__D.PRM 
 IMG-HH-ALOS242498xxxx-220406-FBDR1.1__D.raw
 IMG-HH-ALOS247052xxxx-230208-FBDR1.1__D.PRM 
 IMG-HH-ALOS247052xxxx-230208-FBDR1.1__D.raw 
 Then I use [p2p_processing.csh ALOS2 IMG-HH-ALOS242498xxxx-220406-FBDR1.1__D IMG-HH-ALOS247052xxxx-230208-FBDR1.1__D config.txt], at the top-level directory, starting from stage 2, skipping stage 3,4,5,6 as an experiment. 
  But it gives the following errors:
<img width="1644" alt="Screenshot 2023-04-11 at 10 07 12 AM" src="https://user-images.githubusercontent.com/103137212/231237599-ee0b04c5-3e0d-42c7-a6f2-19df2581b0d6.png">
Here are all the files in my raw folder:
<img width="1724" alt="Screenshot 2023-04-11 at 10 08 55 AM" src="https://user-images.githubusercontent.com/103137212/231237966-e324d034-79b9-4e55-8d5a-89bb845e23bc.png">
And my SLC folder:
<img width="1809" alt="Screenshot 2023-04-11 at 10 09 28 AM" src="https://user-images.githubusercontent.com/103137212/231238090-b07bf27e-9d63-4753-9110-07fe61c38c86.png">
  I wonder what the problem may be. I appreciate it!!

Update:
I tried keeping all the file names unchanged after merging but it's still not working.

**Comment by Xiaohua-Eric-Xu** [comment]
ALOS_merge is written to merge ALOS-1 raw data. For ALOS-2, you could either process to interferogram and then stitch together by calculating number of lines offset along azimuth, or coregister the radar image and then run stitch_tops. I never tried the latter before so am not sure whether it work or not.

**Reply by x3zou** [reply]
Hi @Xiaohua-Eric-Xu 
  Thank you very much for the clarification. I prefer to use stitch_tops before processing so that there will be fewer uncertainties in estimating the ramps for each frame. 
  My ultimate goal is to calculate the RNG and AZO offset between the two dates within the merged frames. So I used stitch_tops but it turned out that the offsets in xorr.dat were very high. So there are a few things I would appreciate your advice:
  1. When I was using **[stitch_tops stem.in output_stem]** I was not sure what the stem.in should exactly be. For example, my stem.in for date 220406 like this:
<img width="372" alt="Screenshot 2023-04-13 at 4 42 40 PM" src="https://user-images.githubusercontent.com/103137212/231905653-cbeb5aee-f7b7-4f1c-bfdd-af19d212c745.png">
  And my stem.in for the the date 230208 is like this:
<img width="279" alt="Screenshot 2023-04-14 at 3 01 04 PM" src="https://user-images.githubusercontent.com/103137212/232162265-c918c54c-76a1-4c14-b4db-5d6a00229a32.png">

  I wonder if it is in the correct form.


 2. After generating my merged files **IMG-HH-220406-merge** and I**MG-HH-230208-merge**, I just used **[align.csh IMG-HH-220406-merge IMG-HH-230208-merge]** to generate xcorr.dat. I wonder if that is the right way to calculate the offsets.

Here is my raw folder after running the stitch_tops and align.csh:


<img width="1263" alt="Screenshot 2023-04-13 at 4 51 17 PM" src="https://user-images.githubusercontent.com/103137212/231906697-b0e1f7d4-05f6-49f1-84c6-1133a8213641.png">

Regards,
Xiaoyu

**Reply by x3zou** [reply]
Hi @Xiaohua-Eric-Xu 
  I also tried slc2amp.csh to plot the merged 220406 frames and 230208 frames and it seems that they are continuous, as shown below. So the merging should be fine (?). 
<img width="461" alt="Screenshot 2023-04-15 at 15 59 55" src="https://user-images.githubusercontent.com/103137212/232256579-6cc6ba6d-55ed-403d-b2a8-b9d3eb20cfb1.png">
<img width="466" alt="Screenshot 2023-04-15 at 16 01 39" src="https://user-images.githubusercontent.com/103137212/232256654-04f604b4-ef45-46f8-8593-01e04c03d132.png">
I'm still puzzled by the output of offsets. I used align.csh to align the 220406 and 230208 merged files, and then ran xcorr command again, with -noshift enforced, to generate freq_xcorr.dat. But the output freq_xcorr.dat has a very high value in both range and azimuth relative offsets. So I wonder if my calculation process is OK.
Thanks

**Reply by Xiaohua-Eric-Xu** [reply, marked as answer]
> 1. When I was using **[stitch_tops stem.in output_stem]** I was not sure what the stem.in should exactly be. For example, my stem.in for date 220406 like this:
> I wonder if it is in the correct form.

Feels correct to me.

> 2. After generating my merged files **IMG-HH-220406-merge** and I**MG-HH-230208-merge**, I just used **[align.csh IMG-HH-220406-merge IMG-HH-230208-merge]** to generate xcorr.dat. I wonder if that is the right way to calculate the offsets.
> 

If you are dealing with ALOS-2 data, then probably want to use `p2p_processing.csh` by setting the config file with `proc_stage = 2` and `skip_stage = 3,4,5,6`.

> I'm still puzzled by the output of offsets. I used align.csh to align the 220406 and 230208 merged files, and then ran xcorr command again, with -noshift enforced, to generate freq_xcorr.dat. But the output freq_xcorr.dat has a very high value in both range and azimuth relative offsets. So I wonder if my calculation process is OK.

When you run the above p2p_processing.csh, it only does a crude xcorr with ~512 outputs. To run with higher resolution, try `make_a_offset.csh`. Then you could change the script to output range grids as well.

**Reply by x3zou** [reply]
Hi @Xiaohua-Eric-Xu , 
  I followed your steps and it is now slowly writing up a higher-resolution freq_xcorr.dat that looks much more desirable to me (although it seems to be taking an immense amount of time). I will let you know if I encounter issues with that later on. I really appreciate your help!

**Reply by Xiaohua-Eric-Xu** [reply]
Great, do share your results so we have a ref on what to do in the scenario you described above.
